scripts/o3de/o3de/repo.py

```diff
@@ -79,6 +70,79 @@ def process_add_o3de_repo(file_name: str or pathlib.Path,
             download_file_result = utils.download_file(parsed_uri, cache_file, True)
             if download_file_result != 0:
                 return download_file_result
+    return 0
+
+def validate_remote_repo(repo_uri: str, validate_contained_objects: bool = False) -> bool:
+    manifest_uri = get_repo_manifest_uri(repo_uri)
+
+    if not manifest_uri:
+        return False
+
+    cache_file = download_repo_manifest(manifest_uri)
+
+    if not cache_file:
+        return False
+
+    if not validation.valid_o3de_repo_json(cache_file):
+        return False
+
+    if validate_contained_objects:
+        repo_data = {}
+        with file_name.open('r') as f:
+            try:
+                repo_data = json.load(f)
+            except json.JSONDecodeError as e:
+                return False
+
+        if download_object_manifests(repo_data) != 0:
+            return False
+
+        gem_set = get_gem_json_paths_from_cached_repo(repo_uri)
+        for gem_json in gem_set:
+            if not validation.valid_o3de_gem_json(gem_json):
+                return False
+
+        project_set = get_project_json_paths_from_cached_repo(repo_uri)
+        for project_json in project_set:
+            if not validation.valid_o3de_project_json(project_json):
+                return False
+
+        template_set = get_template_json_paths_from_cached_repo(repo_uri)
+        for template_json in template_set:
+            if not validation.valid_o3de_template_json(template_json):
+                return False
+
+    return True
+
+def process_add_o3de_repo(file_name: str or pathlib.Path,
+                          repo_set: set) -> int:
+    file_name = pathlib.Path(file_name).resolve()
+    if not validation.valid_o3de_repo_json(file_name):
+        logger.error(f'Repository JSON {file_name} could not be loaded or is missing required values')
+        return 1
+    cache_folder = manifest.get_o3de_cache_folder()
+
+    repo_data = {}
+    with file_name.open('r') as f:
+        try:
+            repo_data = json.load(f)
+        except json.JSONDecodeError as e:
+            logger.error(f'{file_name} failed to load: {str(e)}')
+            return 1
+
+    with file_name.open('w') as f:
+        try:
+            time_now = datetime.now()
+            # Convert to lower case because AM/PM is capitalized
+            time_str = time_now.strftime('%d/%m/%Y %I:%M%p').lower()
```

Comment:
For timestamp, its really important to at least include the time zone information '%Z' in the string